packages/frontend/src/state/lp/hooks.ts

```diff
@@ -135,6 +136,130 @@ export const useOpenPositionDeposit = () => {
   return openPositionDeposit
 }
 
+// Close position with flashloan
+export const useFlashClosePosition = () => {
+  const address = useAtomValue(addressAtom)
+  const controllerHelperContract = useAtomValue(controllerHelperHelperContractAtom)
+  const controllerContract = useAtomValue(controllerContractAtom)
+  const handleTransaction = useHandleTransaction()
+  const getDebtAmount = useGetDebtAmount()
+  const getVault = useGetVault()
+  const getPosition = useGetPosition()
+  const getExactIn = useGetExactIn()
+  const getExactOut = useGetExactOut()
+  const getDecreaseLiquidity = useGetDecreaseLiquidity()
+  const isWethToken0 = useAtomValue(isWethToken0Atom)
+  const index = useAtomValue(indexAtom)
+  const normFactor = useAtomValue(normFactorAtom)
+  const flashClosePosition = useAppCallback(
+    async (
+      vaultId: number,
+      liquidityPercentage: number,
+      burnPercentage: number,
+      withdrawAmount: number,
+      burnExactRemoved: boolean,
+      slippage: number,
+      onTxConfirmed?: () => void,
+    ) => {
+      const vaultBefore = await getVault(vaultId)
+      const uniTokenId = vaultBefore?.NFTCollateralId
+      const position = await getPosition(uniTokenId)
+
+      if (
+        !controllerContract ||
+        !controllerHelperContract ||
+        !address ||
+        !position ||
+        !vaultBefore ||
+        !vaultBefore.shortAmount
+      )
+        return
+
+      const shortAmount = fromTokenAmount(vaultBefore.shortAmount, OSQUEETH_DECIMALS)
+
+      // Get current LP positions
+      const { amount0, amount1 } = await getDecreaseLiquidity(
+        uniTokenId,
+        position.liquidity,
+        0,
+        0,
+        Math.floor(Date.now() / 1000 + 86400),
+      )
+      const wPowerPerpAmountInLP = isWethToken0 ? amount1 : amount0
+
+      const amountToLiquidate = new BigNumber(wPowerPerpAmountInLP).times(liquidityPercentage)
+      const amountToBurn = shortAmount.times(burnPercentage)
+      const limitEth = await (amountToLiquidate.gt(amountToBurn)
+        ? getExactIn(amountToLiquidate.minus(amountToBurn), true)
```

Comment:
right now, this doesn't take the pool fee in and do any autorouting, so we have to restrict this functionality to to be only the 0.3% pool